upup/pkg/fi/cloudup/openstack/instance.go

```diff
@@ -65,10 +69,45 @@ func IsPortInUse(err error) bool {
 	return false
 }
 
+// waitForStatusActive uses gopherclouds WaitFor() func to determine when the server becomes "ACTIVE".
+//
+// The function will immediately fail if the server transistions into the status "ERROR"
+// and will result in a timeout when not reaching status "ACTIVE" in time.
+func waitForStatusActive(c OpenstackCloud, serverID string) error {
+	return gophercloud.WaitFor(int(activeTimeout.Seconds()), func() (bool, error) {
+		server, err := servers.Get(c.ComputeClient(), serverID).Extract()
+		if err != nil {
+			return false, err
+		}
+
+		if server.Status == errorStatus {
+			return false, fmt.Errorf("unable to create server: %v", server.Fault)
+		}
+
+		if server.Status == activeStatus {
+			return true, nil
+		}
+
+		return false, nil
+	})
+}
```

Comment:
@zetaab this `WaitFor()` will now stop immediately when a server transitions into status `ERROR` and i have increased the timeout to 120 seconds (like in the former `WaitForStatusActive()` func in [openstacktasks/instance.go](https://github.com/kubernetes/kops/pull/15102/files#diff-948b4589cbdb93053f2dbe3ee8a1e710d3d171030bcfe2b4efdbba88ad8c80eeL99-L102), which AFAIK is not used anywhere.

Reply:
i will also try to create a test for this, but i struggle a bit with the mockcloud implementation right now